torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -149,6 +149,53 @@ inline Tensor adaptive_avg_pool3d(const Tensor& input,
    return torch::adaptive_avg_pool3d(input, options.output_size());
 }
 
+// ============================================================================
+
+inline std::vector<int64_t> _unpool_output_size(const Tensor& input,
+  const IntArrayRef& kernel_size, const IntArrayRef& stride,
+  const IntArrayRef& padding, const c10::optional<IntArrayRef>& output_size) {
+  auto input_size = input.sizes();
+  std::vector<int64_t> default_size;
+  for (size_t d = 0; d < kernel_size.size(); d++) {
+    default_size.push_back((input_size[d + 2] - 1) * stride[d] +
+                            kernel_size[d] - 2 * padding[d]);
+  }
+  if (!output_size) {
+    return default_size;
+  } else {
+    std::vector<int64_t> output_size_;
+    if (output_size->size() == kernel_size.size() + 2) {
+      output_size_ = {output_size->begin() + 2, output_size->end()};
```

Comment:
To better match the Python version, maybe we should write
```suggestion
      output_size_ = output_size->slice(2);
```

Reply:
output_size_ = output_size->slice(2).vec();

Reply:
`output_size_ = output_size->slice(2).vec();` looks great. Do you plan to change it?